src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2398,53 +2399,115 @@ def delete_ssl_cert(cmd, resource_group_name, certificate_thumbprint):
     raise CLIError("Certificate for thumbprint '{}' not found".format(certificate_thumbprint))
 
 
-def import_ssl_cert(cmd, resource_group_name, name, key_vault, key_vault_certificate_name):
-    Certificate = cmd.get_models('Certificate')
+def import_ssl_cert(cmd, resource_group_name, name, certificate=None, key_vault_certificate_name=None,
+                    key_vault=None):
+    Certificate_Model = cmd.get_models('Certificate')
     client = web_client_factory(cmd.cli_ctx)
     webapp = client.web_apps.get(resource_group_name, name)
     if not webapp:
-        raise CLIError("'{}' app doesn't exist in resource group {}".format(name, resource_group_name))
+        err_msg = "'{}' app doesn't exist in resource group {}".format(name, resource_group_name)
+        raise ResourceNotFoundError(err_msg)
     server_farm_id = webapp.server_farm_id
     location = webapp.location
-    kv_id = None
-    if not is_valid_resource_id(key_vault):
-        kv_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_KEYVAULT)
-        key_vaults = kv_client.vaults.list_by_subscription()
-        for kv in key_vaults:
-            if key_vault == kv.name:
-                kv_id = kv.id
-                break
+
+    cert_subscription = ''
+    if certificate and key_vault_certificate_name:
+        no_two_certs = "Provide only one certificate name, use \'--certificate-name\'."
+        logger.warning(no_two_certs)
+        return
+    if certificate:
+        if is_valid_resource_id(certificate):
+            cert_id_parts = parse_resource_id(certificate)
+            cert = cert_id_parts['name']
+            cert_subscription = cert_id_parts['subscription']
+        else:
+            cert = certificate
+    elif key_vault_certificate_name:
```

Comment:
key_vault_certificate_name is currently a required property on the command so we will always have this value.
More than likely users will use this for both ASV certs & the new ASD certs. The code should be able to parse this correctly & handle this.